src/v/kafka/server/group.cc

```diff
@@ -1447,6 +1447,96 @@ ss::future<error_code> group::remove() {
     co_return error_code::none;
 }
 
+ss::future<>
+group::remove_topic_partitions(const std::vector<model::topic_partition>& tps) {
+    std::vector<std::pair<model::topic_partition, offset_metadata>> removed;
+    for (const auto& tp : tps) {
+        _pending_offset_commits.erase(tp);
+        if (auto offset = _offsets.extract(tp); offset) {
+            removed.emplace_back(
+              std::move(offset.key()), std::move(offset.mapped()));
+        }
+    }
+
+    // if no members and no offsets
+    if (
+      in_state(group_state::empty) && _pending_offset_commits.empty()
+      && _offsets.empty()) {
+        vlog(
+          klog.debug,
+          "Marking group {} as dead at {} generation",
+          _id,
+          generation());
+
+        set_state(group_state::dead);
+    }
+
+    if (removed.empty()) {
+        co_return;
+    }
+
+    _offsets.rehash(0);
+    _pending_offset_commits.rehash(0);
+
+    // build offset tombstones
+    storage::record_batch_builder builder(
+      raft::data_batch_type, model::offset(0));
+
+    // create deletion records for offsets from deleted partitions
+    for (auto& offset : removed) {
+        vlog(
+          klog.trace, "Removing offset for group {} tp {}", _id, offset.first);
+
+        group_log_record_key key{
+          .record_type = group_log_record_key::type::offset_commit,
+          .key = reflection::to_iobuf(group_log_offset_key{
+            _id,
+            offset.first.topic,
+            offset.first.partition,
+          }),
+        };
+
+        builder.add_raw_kv(reflection::to_iobuf(std::move(key)), std::nullopt);
```

Comment:
very nice.

Reply:
I don't see the hydration step where it ignores std::nullopt or rather removes it from the map.

Reply:
`recover_partition` may be what you're looking for?

Reply:
i looked at it, but *something* determines what groups to be removed. 

can you poste link to the line of code that removes it. 

i'm looking for `if value == nullopt; map.erase( key ) `

Reply:
how about this?

https://github.com/vectorizedio/redpanda/blob/dev/src/v/kafka/server/group_manager.cc#L334

Reply:
ah!!! tip-top! thank you.